scripts/openyurt-deployer/main_test.go

```diff
@@ -0,0 +1,177 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestUnmarshallingJsonFile(t *testing.T) {
+	// Create a temporary file with JSON content
+	tempFile, err := ioutil.TempFile("", "test.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(tempFile.Name())
+
+	// Define your sample JSON content
+	jsonContent := `{
+		"master": "username@masterip",
+		"workers": {
+			"cloud": [
+				"username@cloudip"
+			],
+			"edge": [
+				"username@edgeip"
+			]
+		}
+	}`
+
+	// Write the JSON content to the temporary file
+	err = ioutil.WriteFile(tempFile.Name(), []byte(jsonContent), 0644)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Test the readAndUnMarshall function with the temporary file
+	result, err := readAndUnMarshall(tempFile.Name())
+	if err != nil {
+		t.Fatalf("Unexpected error: %s", err)
+	}
+
+	t.Logf("Unmarshall: %s", result)
+	// Define the expected result based on your JSON structure
+	expected := NodesInfo{
+		// Initialize the fields based on your JSON structure
+		Master: "username@masterip",
+		Workers: Workers{
+			Cloud: []string{"username@cloudip"},
+			Edge:  []string{"username@edgeip"},
+		},
+	}
+
+	t.Logf("expected res: %s", expected)
+	// Compare the result with the expected value
+	if !jsonEqual(result, expected) {
+		t.Errorf("Expected %v, got %v", expected, result)
+	}
+}
+
+// jsonEqual checks if two JSON objects are equal.
+func jsonEqual(a, b interface{}) bool {
+	ajson, err := json.Marshal(a)
+	if err != nil {
+		return false
+	}
+
+	bjson, err := json.Marshal(b)
+	if err != nil {
+		return false
+	}
+
+	return string(ajson) == string(bjson)
+}
+
+func TestParsingNodeDependencyVersion(t *testing.T) {
+
+	// Mock NodesInfo
+	mockNodesInfo := NodesInfo{
+		Master: "runner@127.0.0.1",
+		// Workers: Workers{
+		// 	Cloud: []string{"cloud@host"},
+		// 	Edge:  []string{"edge@host"},
+		// },
+	}
+
+	// Define a table of criteria
+	criteriaTable := map[string]string{
+		"Golang":     "1.19.10",
+		"containerd": "1.6.18",
+		"runc":       "1.1.4",
+		"CNI":        "1.2.0",
```

Comment:
These values have to be updated upon every version upgrade. I don't think it's convenient to have them hardcoded. we can have a separate config file (or better: use the versions from the existing config files)